core/model/modx/processors/resource/sort.class.php

```diff
@@ -62,6 +67,12 @@ public function process() {
         $sorted = $this->sort();
         if ($sorted !== true) return $this->failure($sorted);
 
+        if($this->getProperty('source_type')==="modResource"){
+            $resourceAfterSort = $this->modx->getObject('modResource', array('id' => $this->getProperty('source_pk')));
+            $newParentID = $resourceAfterSort->get('parent');
```

Comment:
Same as above, if for some weird reason the resource can't be loaded, this would cause a fatal error "method get on null".